source/extensions/filters/common/expr/context.cc

```diff
@@ -39,242 +39,423 @@ convertHeaderEntry(Protobuf::Arena& arena,
 
 namespace {
 
+const absl::flat_hash_map<absl::string_view, SslExtractor>& getSslExtractors() {
+  static const auto* extractors = new absl::flat_hash_map<absl::string_view, SslExtractor>{
```

Comment:
prefer to use smart pointers instead (e.g. std::make_unique<absl::flat_hash_map<absl::string_view, SslExtractor>>() depending on the ownership model) here and in all `new` occurrences below

Reply:
You could use `ConstSingleton` with an object that has the map. This is a common way to store static tables.

Reply:
I switched it to using `CONSTRUCT_ON_FIRST_USE`